crates/ruff_python_formatter/src/type_param/type_param_clause.rs

```diff
@@ -0,0 +1,19 @@
+use crate::builders::PyFormatterExtensions;
+use crate::context::PyFormatContext;
+use crate::prelude::*;
+use ruff_formatter::{Format, FormatResult};
+use ruff_python_ast::TypeParam;
+use ruff_text_size::TextSize;
+
+pub(crate) struct FormatTypeParamsClause<'a> {
+    pub(crate) sequence_end: TextSize,
+    pub(crate) type_params: &'a Vec<TypeParam>,
```

Comment:
It's generally preferred to pass slices rather than references to vectors. 

Rust passes a pointer to the vector's address if you pass a vector by reference. That result is that accessing the first element requires two dereferences:

1. Derefence the `&Vec` to get `Vec``
2. Dereference `start` pointer to get the first element


Rust avoids the first dereference if you pass the vec as a slice because it directly passes the start and end positions (omitting the Vec's capacity). This removes the need for the first dereference.

```suggestion
    pub(crate) type_params: &'a [TypeParam],
```

Reply:
Another reason is that `&[T]` is in a way a superclass to `&Vec<T>` (`Vec<T>` derefs to `[T]`) and allows passing e.g. slices of a vec too.

Reply:
Thanks for the additional knowledge!